OpenRA.Mods.Common/Traits/Sellable.cs

```diff
@@ -63,9 +66,14 @@ public void Sell(Actor self)
 			foreach (var ns in self.TraitsImplementing<INotifySold>())
 				ns.Selling(self);
 
-			var makeAnimation = self.TraitOrDefault<WithMakeAnimation>();
-			if (makeAnimation != null)
-				makeAnimation.Reverse(self, new Sell(self), false);
+			if (!info.SkipMakeAnimation)
+			{
+				var makeAnimation = self.TraitOrDefault<WithMakeAnimation>();
+				if (makeAnimation != null)
+					makeAnimation.Reverse(self, new Sell(self), false);
+				else
+					self.QueueActivity(false, new Sell(self));
```

Comment:
Duplicating this `QueueActivity` isn't great.  If you return after the `makeAnimation.Reverse` then you can have just one at the end of the method instead.